CFC_DataCollector/trip_generator/tour_model.py

```diff
@@ -0,0 +1,229 @@
+import random
+from util import *
+
+class Location(object):
+
+    def __init__(self, name):
+        self.successors = [ ]
+        self.days_of_week = [ ]
+        self.name = name
+        self.times_allowed_one = 0
+        self.times_allowed_two = 0
+        self.count = 0
+        self.count_down = 0
+        self.successors_counter = Counter()
+
+    def add_to_successors(self, loc, weight=None):
+        if isinstance(loc, dict):
+            print loc
+            for location, w in loc.iteritems():
+                self.successors_counter[location] = w
+        elif type(loc) == Location:
+            self.successors_to_probs[loc] = weight
+        else:
+            print "You can not input the location as type %s, please use a dictionary or location" % type(loc)
+
+        # loc.times_allowed_one += 1
+        # loc.times_allowed_two += 1
+        # loc.count += 1
+
+    def get_successors(self):
+        return self.successors
+
+    def visit(self):
+        self.times_allowed_one -= 1
+
+    def __repr__(self):
+        return self.name
+
+    def __str__(self):
+        return self.name
+
+class TourModel(object):
+
+    def __init__(self, home, name):
+        self.home = home
+        self.name = name
+        self.all_locs = set( )
+        for loc in self.home.get_successors():
+            loc.count -= 1
+
+    def get_tour_model_from(self, place):
+        #return _dfs_search(self, place)
+        tour_model = [ ]
+        orig_place = place 
+        curr_node = place
+        tour_model.append(curr_node)
+        start = True
+        while (curr_node != orig_place) or start:
+            start = False
+            curr_node = get_next_node(curr_node) ## Do a random walk around the FSM, fix this later??
+            tour_model.append(curr_node)
+        return tour_model
+
+    def get_all_tour_models(self):
+        tour_models = [ ]
+        for place in self.home.get_successors():
+            tour_models.append(self.get_tour_model_from(place))
+        return tour_models
+        
+
+def get_next_node(place):
+    # sucessor = choose_random_successor(place.get_successors())
+    # return sucessor
+    return sampleFromCounter(place.successors_counter)
+
+def choose_random_successor(successors):
+    num_successors = len(successors)
+    index = random.randint(0, num_successors - 1)
+    return successors[index]
+
+
+def _dfs_search(TM, starting_point):
```

Comment:
Aren't there existing graph libraries for python that you could use? I think that Naomi has some pointers...